lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -638,10 +674,27 @@ JavaScriptCompiler.prototype = {
         '(', lookup,
         (helper.paramsInit ? ['),(', helper.paramsInit] : []), '),',
         '(typeof helper === ', this.aliasable('"function"'), ' ? ',
-        this.source.functionCall('helper', 'call', helper.callParams), ' : helper))'
+        this.helperFunctionCall('helper', helper, null), ' : helper))'
     ]);
   },
 
+  helperFunctionCall: function(helperName, helperOptions, splatMap) {
+    if (splatMap) {
+      let splatMapObj = {};
+      for (let i = 0, l = splatMap.length; i < l; ++i) {
+        splatMapObj[splatMap[i]] = 1;
+      }
+
+      let argsWithSplatMap = helperOptions.params.slice();
+      argsWithSplatMap.push(this.objectLiteral(splatMapObj));
```

Comment:
`push` mutates. `concat` returns a new array. Either way the goal seems to be to create a new array to avoid mutating `helperOptions.params`. Though I suppose VMs might optimize `a.concat(b)` in ways they can't `a.slice().push(b)`.